packages/react/src/primitives/Breadcrumbs/BreadcrumbItem.tsx

```diff
@@ -0,0 +1,59 @@
+import * as React from 'react';
+import classNames from 'classnames';
+
+import { classNameModifierByFlag } from '../shared/utils';
+import { ComponentClassNames } from '../shared/constants';
+import {
+  BaseBreadcrumbItemProps,
+  BreadcrumbsItemProps,
+  ForwardRefPrimitive,
+  Primitive,
+} from '../types';
+import { View } from '../View';
+import { BreadcrumbsContext } from './BreadcrumbsContext';
+
+const BreadcrumbItemPrimitive: Primitive<BreadcrumbsItemProps, 'li'> = (
+  { className, children, isCurrent, ...rest },
+  ref
+) => {
+  const { separator } = React.useContext(BreadcrumbsContext);
+  const componentClasses = classNames(
+    ComponentClassNames.BreadcrumbsItem,
+    classNameModifierByFlag(
+      ComponentClassNames.BreadcrumbsItem,
+      'current',
+      isCurrent
+    ),
+    className
+  );
+  const value = React.useMemo(() => {
+    return { isCurrent };
+  }, [isCurrent]);
+
+  return (
+    <View as="li" className={componentClasses} ref={ref} {...rest}>
+      <BreadcrumbsContext.Provider value={value}>
```

Comment:
Asked abt `isCurrent` in another comment, but maybe that would be resolved by having a context that is specific to  `BreadcrumbItem`?